spec/support/example_app_helper.rb

```diff
@@ -0,0 +1,37 @@
+require 'fileutils'
+
+module ExampleAppHelper
+  def setup_example_app
+    FileUtils.cd(tmp_path) do
+      `rails new #{app_name} -d postgresql --skip-action-mailer --skip-action-mailbox --skip-action-cable --skip-sprockets --skip-listen --skip-javascript --skip-turbolinks --skip-system-test --skip-test-unit --skip-bootsnap --skip-spring`
+    end
+
+    File.open("#{example_app_path}/Gemfile", 'a') do |f|
+      f.puts "gem 'good_job'"
+    end
+  end
+
+  def run_in_example_app(*args)
+    FileUtils.cd(example_app_path) do
+      system(*args) || raise("Command #{args} failed")
+    end
+  end
+
+  def remove_example_app
+    FileUtils.rm_rf(example_app_path)
+  end
+
+  def example_app_path
+    File.join(tmp_path, app_name)
+  end
+
+  def tmp_path
+    @_tmp_path ||= File.join(File.dirname(__FILE__), '..')
+  end
+
+  def app_name
+    'example_app'
+  end
+end
+
+RSpec.configure { |c| c.include ExampleAppHelper, type: :generator }
```

Comment:
```ruby
  def setup_example_app
    root_path = example_app_path.join('..')
    FileUtils.cd(root_path) do
      `rails new #{app_name} -d postgresql --skip-action-mailer --skip-action-mailbox --skip-action-cable --skip-sprockets --skip-listen --skip-javascript --skip-turbolinks --skip-system-test --skip-test-unit --skip-bootsnap --skip-spring`
    end

    File.open(example_app_path.join("Gemfile"), 'a') do |f|
      f.puts "gem 'good_job'"
    end
  end

  #...

  def example_app_path
    Rails.root.join('..', app_name)
  end

  def app_name
    'example_app'
  end
end

RSpec.configure { |c| c.include ExampleAppHelper, type: :generator }
```

Reply:
I made a few minor changes but yeah, we got rid of `tmp_path` 😄